C/plugins/north/http-north/plugin.cpp

```diff
@@ -0,0 +1,253 @@
+				"\"type\": \"string\", " \
+				"\"default\": \"http://localhost:6683/sensor-reading\" }, " \
+			"\"HttpTimeout\": { " \
+				"\"description\": \"Timeout in seconds for the HTTP operations with the HTTP Connector Relay\", " \
+				"\"type\": \"integer\", \"default\": \"10\" }, " \
+			"\"verifySSL\": { " \
+        			"\"description\": \"Verify SSL certificate\", " \
+				"\"type\": \"boolean\", \"default\": \"False\" }, " \
+			"\"applyFilter\": { " \
+        			"\"description\": \"Whether to apply filter before processing the data\", " \
+				"\"type\": \"boolean\", \"default\": \"False\" }, " \
+			"\"filterRule\": { " \
+				"\"description\": \"JQ formatted filter to apply (applicable if applyFilter is True)\", " \
+				"\"type\": \"string\", \"default\": \".[]\" }"
+
+/**
+ * The HTTP north plugin interface
+ */
+extern "C" {
+
+/**
+ * The C API plugin information structure
+ */
+static PLUGIN_INFORMATION info = {
+	"http",			// Name
+	"1.0.0",		// Version
+	0,			// Flags
+	PLUGIN_TYPE_NORTH,	// Type
+	"1.0.0",		// Interface version
+	PLUGIN_DEFAULT_CONFIG   // Configuration
+};
+
+static const map<const string, const string> plugin_configuration = {
+					{
+						"PLUGIN",
+						string(PLUGIN_DEFAULT_CONFIG)
+					}
+				 };
+
+/**
+ * HTTP Server connector info
+ */
+typedef struct
+{
+	string		proto;
+	HttpSender	*sender;  // HttpSender is the base class for SimpleHttp and SimpleHttp classes
+				  // and sendRequest is a virtual method of HttpSender class
+	string		path;
+} CONNECTOR_INFO;
+
+uint32_t sendToServer(const vector<Reading *>& readings, CONNECTOR_INFO *connInfo);
+const vector<pair<string, string>> createMessageHeader();
+const string& getReadingString(const Reading& reading);
+
+/**
+ * Return the information about this plugin
+ */
+PLUGIN_INFORMATION *plugin_info()
+{
+	return &info;
+}
+
+/**
+ * Return default plugin configuration:
+ * plugin specific and types_id
+ */
+const map<const string, const string>& plugin_config()
+{
+	return plugin_configuration;
+}
+
+/**
+ * Initialise the plugin with configuration.
+ *
+ * This funcion is called to get the plugin handle.
+ */
+PLUGIN_HANDLE plugin_init(map<string, string>&& configData)
+{
+	Logger::getLogger()->info("http-north C plugin: %s", __FUNCTION__);
+
+	/**
+	 * Handle the HTTP(S) parameters here
+	 */
+	ConfigCategory configCategory("cfg", configData["GLOBAL_CONFIGURATION"]);
+	string url = configCategory.getValue("URL");
+	unsigned int timeout = atoi(configCategory.getValue("HttpTimeout").c_str());
+
+	/**
+	 * Extract host, port, path from URL
+	 */
+	size_t findProtocol = url.find_first_of(":");
+	string protocol = url.substr(0,findProtocol);
+
+	string tmpUrl = url.substr(findProtocol + 3);
+	size_t findPort = tmpUrl.find_first_of(":");
+	string hostName = tmpUrl.substr(0, findPort);
+
+	size_t findPath = tmpUrl.find_first_of("/");
+	string port = tmpUrl.substr(findPort + 1 , findPath - findPort -1);
+	string path = tmpUrl.substr(findPath);
```

Comment:
can this be a utility method? if http sender lib always needs us to split url and find host, port and uri path